plugins/inputs/netdata/netdata.go

```diff
@@ -0,0 +1,291 @@
+  ## to netdata we get measurement for each second. This value is capped by "update every" config of netdata.
+  ## For example you can set
+  ## points = 20
+  ## interval = "10m"
+  ## to get measurement every 30 seconds (and polling netdata every 10 minutes)
+  ## default to 10
+  points = 10
+
+  ## Grouping method if multiple collected values are to be grouped in order to return fewer points.
+  ## methods supported "min", "max", "average", "sum", "incremental-sum"
+  ## cf : https://github.com/firehol/netdata/wiki/REST-API-v1
+  ## default to average
+  group = "average"
+
+  ## Should measurement be prefixed by the netdata hostname ?
+  ## Ex : netdata.host1.apps.cpu (with hostname),  netdata.apps.cpu (without hostname)
+  ## default to true
+  prefixChartIDByHostname = true
+
+  ## Netdata servers
+  [[inputs.netdata.servers]]
+    url="http://netdata-host:19999"
+
+  [[inputs.netdata.servers]]
+    url="https://netdata-host2"
+    basicAuthUsername="user"
+    basicAuthPassword="pass"
+`
+
+type Netdata struct {
+	Servers                 []Server
+	Points                  int
+	Group                   string
+	PrefixChartIDByHostname bool
+
+	client *http.Client
+}
+
+type Server struct {
+	Url               string
+	BasicAuthUsername string
+	BasicAuthPassword string
+	lastGatherTime    int64
+	currGatherTime    int64
+}
+
+type NestedError struct {
+	Err       error
+	NestedErr error
+}
+
+func (ne NestedError) Error() string {
+	return ne.Err.Error() + ": " + ne.NestedErr.Error()
+}
+
+func Errorf(err error, msg string, format ...interface{}) error {
+	return NestedError{
+		NestedErr: err,
+		Err:       fmt.Errorf(msg, format...),
+	}
+}
+
+type ResponseCharts struct {
+	Hostname string           `json:"hostname"`
+	Charts   map[string]Chart `json:"charts"`
+}
+
+type Chart struct {
+	ID         string                    `json:"id"`
+	Type       string                    `json:"type"`
+	Family     string                    `json:"family"`
+	DataURL    string                    `json:"data_url"`
+	Dimensions map[string]ChartDimension `json:"dimensions"`
+}
+
+type ChartDimension struct {
+	Name string `json:"name"`
+}
+
+type ResponseData struct {
+	DimensionNames []string   `json:"dimension_names"`
+	Result         DataResult `json:"result"`
+}
+
+type DataResult struct {
+	Data [][]interface{} `json:"data"`
+}
+
+func init() {
+	inputs.Add("netdata", func() telegraf.Input {
+		return &Netdata{
+			Points: 10,
+			Group:  "average",
+			PrefixChartIDByHostname: true,
+		}
+	})
+}
+
+func (n *Netdata) Description() string {
+	return description
+}
+
+func (n *Netdata) SampleConfig() string {
+	return sampleConfig
+}
+
+func (n *Netdata) Gather(acc telegraf.Accumulator) error {
+	var wg sync.WaitGroup
+
+	if n.client == nil {
+		tr := &http.Transport{
+			ResponseHeaderTimeout: time.Duration(3 * time.Second),
+		}
+		n.client = &http.Client{
+			Transport: tr,
+			Timeout:   time.Duration(4 * time.Second),
+		}
```

Comment:
I think this should also be done in the `Init()` function. Please also check that the resulting client is not `nil` just in case.